### PR TITLE
Fire source control hooks for opened and closed documents

### LINKED ISSUE #1405: VSCode file open and close events don't register in UserAction of %Studio.SourceControl.Base
As a part of the git-source-control package, we would like to be able to have the file open and close events register with the %Studio.SourceControl.Base class. This will allow us to make some of our features compatible in VSCode and not just studio and the management portal.

### PR DESCRIPTION
**Pull request by isc-bsaviano**
This PR fixes #1405

### CODE REVIEW

**Review by isc-rsingh — APPROVED** [review]

**Comment by isc-bsaviano** [comment]
@isc-tleavitt @isc-hwojnick Does someone on your team want to test this change before I merge it?

**Comment by isc-tleavitt** [comment]
I'll volunteer @isc-hwojnick for this.

File: src/extension.ts [hunk 1]
```diff
@@ -1431,6 +1439,22 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
         DocumentContentProvider.getUri(doc, undefined, undefined, undefined, wsFolder.uri)
       );
     }),
+    vscode.window.tabGroups.onDidChangeTabs((e) => {
+      const processUri = (uri: vscode.Uri): void => {
+        if (uri.scheme == FILESYSTEM_SCHEMA) {
```
Comment (gjsjohnmurray) [inline src/extension.ts:1444]: Are the events not wanted for documents coming from `isfs-readonly` folders?

**Review by gjsjohnmurray — COMMENTED** [review]
One inline comment about whether the events should also occur on `isfs-readonly` documents.

File: src/extension.ts (hunk 1 above)
Comment (isc-bsaviano) [reply src/extension.ts:1444]: I don't think so. For example, we only show the source control menu for `isfs` files. What do you think @isc-tleavitt?

**Comment by isc-hwojnick** [comment]
It works great! Thank you guys for the quick turn around!

**Pull request merged**